public/controllers/agent/components/register-agent.js

```diff
@@ -293,10 +298,18 @@ export const RegisterAgent = withErrorBoundary(
 
     setAgentName(event) {
       this.setState({ agentName: event.target.value });
-    }
-
-    setAgentName(event) {
-      this.setState({ agentName: event.target.value });
+      if (/^[a-z0-9-_.]+$/i.test(event.target.value) || event.target.value.length <= 0) {
```

Comment:
Nitpick: If you are going to use the same regular expression more than once, you could store it in a constant to improve maintainability.

Reply:
Solved in https://github.com/wazuh/wazuh-kibana-app/pull/5021/commits/72fcfabb864a08850d43795d989c8cf97909c488